executor.sh

```diff
@@ -25,4 +25,32 @@ echo "--> Copying ${SIDECAR_EXECUTOR_PATH} to sandbox ${MESOS_SANDBOX}"
 executor="${MESOS_SANDBOX}/`basename ${SIDECAR_EXECUTOR_PATH}`"
 cp $SIDECAR_EXECUTOR_PATH $executor
 echo "--> Starting ${executor}"
-exec $executor -logtostderr=true
+
+function cleanup {
+	# Golang template description (details here https://golang.org/pkg/text/template/):
+	# // Iterate over all the container environment variables
+	# {{range $i, $v := .Config.Env}}
+	#     // Select the variable named "TASK_ID"
+	#     {{if eq (index (split $v "=") 0) "TASK_ID"}}
+	#         // ... for the container where it's set to the contents of ${TASK_ID}
+	#         // note the bash string concatenation done by juxtaposition!
+	#         {{if eq (index (split $v "=") 1) "'"${TASK_ID}"'"}}
+	#             // Print the container ID, which is a child of $ (it resolves the root node of the docker inspect output)
+	#             {{$.ID}}
+	#         {{end}}
+	#     {{end}}
+	# {{end}}
+	if [ -n "${TASK_ID}" ]; then
+		# Swallow all errors returned by these calls with `|| true`
+		# to bubble up the status returned by the executor
+		CONTAINER_ID=$(docker ps -q | xargs docker inspect -f '{{range $i, $v := .Config.Env}}{{if eq (index (split $v "=") 0) "TASK_ID"}}{{if eq (index (split $v "=") 1) "'"${TASK_ID}"'"}}{{$.ID}}{{end}}{{end}}{{end}}') || true
```

Comment:
🥇  didn't know you could do this with the `inspect` command!

Reply:
Go templates FTW! Too bad they don't expose more string functions. Writing this using some grep/sed magic would be much shorter.